File: robusta-java-test/src/main/java/InstanceOf.java
public class InstanceOf {

    public static void main(String[] args) {
        test(new Object());
        test("hello world");
        test(new InstanceOf());
    }

    private static void test(Object object) {
        Object[] arr = {new Object()};

        System.out.println("Object: " + (object instanceof Object));
        System.out.println("Object Array: " + (object instanceof Object[]));

        System.out.println("String: " + (object instanceof String));
        System.out.println("String Array: " + (object instanceof String[]));

        System.out.println("InstanceOf: " + (object instanceof InstanceOf));
        System.out.println("InstanceOf Array: " + (object instanceof InstanceOf[]));
    }
}

File: tests/instance_of.rs
use assert_cmd::Command;

#[test]
fn instance_of() {
    let mut command = Command::cargo_bin("robusta").unwrap();

    command
        .env("ROBUSTA_CLASSPATH", "robusta-java-runtime/target/robusta-java-runtime.jar:robusta-java-test/target/robusta-java-test.jar")
        .arg("InstanceOf")
        .assert()
        .success()
        .code(0)
        .stdout("Object: true
Object Array: false
String: false
String Array: false
InstanceOf: false
InstanceOf Array: false
Object: true
Object Array: false
String: true
String Array: false
InstanceOf: false
InstanceOf Array: false
Object: true
Object Array: false
String: false
String Array: false
InstanceOf: true
InstanceOf Array: false
")
        .stderr("");
}
